test/unit/top-app-bar/index.test.js

```diff
@@ -204,6 +204,111 @@ test('#enableRippleOnElement throws error if a native element', () => {
   );
 });
 
+test('when changes from short to fixed the foundation changes', () => {
+  const wrapper = shallow(<TopAppBar short />);
+  const originalFoundation = wrapper.instance().foundation_;
+  wrapper.setProps({fixed: true, short: false});
+  assert.notEqual(wrapper.instance().foundation_, originalFoundation);
+  assert.exists(wrapper.instance().foundation_);
+});
+
+test('when changes from short to fixed the foundation changes', () => {
+  const wrapper = shallow(<TopAppBar short />);
+  const originalFoundation = wrapper.instance().foundation_;
+  wrapper.setProps({fixed: true, short: false});
+  assert.notEqual(wrapper.instance().foundation_, originalFoundation);
+  assert.exists(wrapper.instance().foundation_);
+});
+
+test('when changes from short to standard the foundation changes', () => {
+  const wrapper = shallow(<TopAppBar short />);
+  const originalFoundation = wrapper.instance().foundation_;
+  wrapper.setProps({short: false});
+  assert.notEqual(wrapper.instance().foundation_, originalFoundation);
+  assert.exists(wrapper.instance().foundation_);
+});
+
+test('when changes from short to prominent the foundation changes', () => {
+  const wrapper = shallow(<TopAppBar short />);
+  const originalFoundation = wrapper.instance().foundation_;
+  wrapper.setProps({short: false, prominent: true});
+  assert.notEqual(wrapper.instance().foundation_, originalFoundation);
+  assert.exists(wrapper.instance().foundation_);
+});
+
+test('when changes from short to shortCollpased the foundation changes', () => {
```

Comment:
Shouldn't the foundation not change in this case?

Reply:
From what I understand it should reinitialize. When `init()` is called on `MDCShortTopAppBarFoundation` it registers a resize handler if initialized as a short variant. If it begins life collapsed it would not be necessary to register the resize handler. [[0](https://github.com/material-components/material-components-web/blob/a487783c74a42104a27ed6de42bad9d6d29a0a63/packages/mdc-top-app-bar/short/foundation.js#L44-L56)]  Alternatively the resize handler could simply be deregistered. But I think this would work just as well?


[0] https://github.com/material-components/material-components-web/blob/a487783c74a42104a27ed6de42bad9d6d29a0a63/packages/mdc-top-app-bar/short/foundation.js#L44-L56

Reply:
ah right - this is because you would want the short collapsed to immediately collapse. Good call. Thanks!